src/components/LanguagePicker/index.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Box,
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Kbd,
+  Menu,
+  MenuList,
+  type MenuListProps,
+  type MenuProps,
+  Text,
+  type UseDisclosureReturn,
+  useEventListener,
+} from "@chakra-ui/react"
+
+import { Button } from "@/components/Buttons"
+import { BaseLink } from "@/components/Link"
+
+import MenuItem from "./MenuItem"
+import NoResultsCallout from "./NoResultsCallout"
+import { useLanguagePicker } from "./useLanguagePicker"
+
+type LanguagePickerProps = Omit<MenuListProps, "children"> & {
+  children: React.ReactNode
+  placement?: MenuProps["placement"]
+  handleClose?: () => void
+  menuState?: UseDisclosureReturn
+}
+
+const LanguagePicker = ({
+  children,
+  placement,
+  handleClose,
+  menuState,
+  ...props
+}: LanguagePickerProps) => {
+  const { t, refs, disclosure, filterValue, setFilterValue, filteredNames } =
+    useLanguagePicker(handleClose, menuState)
+  const { inputRef, firstItemRef, noResultsRef, footerRef } = refs
+  const { onClose } = disclosure
+
+  /**
+   * Adds a keydown event listener to focus filter input (\).
+   * @param {string} event - The keydown event.
+   */
+  useEventListener("keydown", (e) => {
+    if (e.key !== "\\") return
+    e.preventDefault()
+    inputRef.current?.focus()
+  })
+
+  return (
+    <Menu isLazy placement={placement} autoSelect={false} {...disclosure}>
+      {children}
+      <MenuList
+        position="relative"
+        overflow="auto"
+        borderRadius="base"
+        py="0"
+        onKeyDown={(e) => {
+          if (e.key === "Tab" || e.key === "\\") {
+            e.preventDefault()
+            ;(e.shiftKey ? inputRef : footerRef).current?.focus()
+          }
+        }}
+        {...props}
+      >
+        {/* Mobile Close bar */}
+        <Flex
+          justifyContent="end"
+          hideFrom="lg" // TODO: Confirm breakpoint after nav-menu PR merged
+          position="sticky"
+          zIndex="sticky"
+          top="0"
+          bg="background.base"
+        >
+          <Button
+            p="4"
+            variant="ghost"
+            alignSelf="end"
+            onClick={() => onClose()}
+          >
+            {t("common:close")}
+          </Button>
+        </Flex>
+
+        {/* Main Language selection menu */}
+        <Box
+          position="relative"
```

Comment:
Not a blocker: wondering if we could treat this as an actual `form`
```suggestion
        <Box
          as="form"
          position="relative"
```

We could have some a11y wins when we submit the form (also, would remove the need of the custom `onKeyDown` handler).